CRM/Extension/Browser.php

```diff
@@ -157,21 +157,31 @@ public function getExtensions() {
 
   /**
    * Get a description of a particular extension.
+   * This function is an optimised form of getExtensions() - it returns the
+   * extension's details as soon as it's found without caching anything.
    *
    * @param string $key
    *   Fully-qualified extension name.
    *
    * @return CRM_Extension_Info|NULL
    */
   public function getExtension($key) {
-    // TODO optimize performance -- we don't need to fetch/cache the entire repo
-    $exts = $this->getExtensions();
-    if (array_key_exists($key, $exts)) {
-      return $exts[$key];
+    if (!$this->isEnabled() || count($this->checkRequirements())) {
+      return NULL;
     }
-    else {
+
+    $remote = $this->_discoverRemote();
```

Comment:
That isn't the point of this PR.

The previous version iterated through all of the extensions, storing them in a variable, before returning the one needed. This doesn't store anything, it just iterates through them and returns the needed one.

When I measured it it performed faster, and will definitely use less memory.

If you can think of further improvements then they'd be welcome in a subsequent PR: this is a localised but notable improvement.